### ISSUE TITLE
FP2: On wakeup brightness of the screen first drops [to minimum?] then after delay gets back to "usable".

### ISSUE THREAD

**Issue by kwah-gh** [ubports/ubuntu-touch#1162]
- Device: FP2
- Channel: 16.04 / rc
- Build: 2019-W24

### Steps to reproduce

When phone is in "standby"
- Press power button to wake it up

### Expected behavior

- Unlock screen is shown so that user can enter unlock phrase

### Actual behavior

- Unlock screen is shown
- Screen is dimmed so that it is not possible to use unlock OSK
- After a significant delay screen brightness gets to a usable level

### Logfiles and additional information

`dmesg` seems to be not updated.

[logcat.txt](https://github.com/ubports/ubuntu-touch/files/3307864/logcat.txt)

**Comment by Ingo-FP-Angel** [comment]
Do you have auto adjust brightness enabled? Is there a difference if that's on or off?

**Comment by kwah-gh** [comment]
Yes, auto adjust brightness is enabled. And yes, it makes a difference.

**Comment by Ingo-FP-Angel** [comment]
I've just checked with auto brightness + brightness slider set to minimum.
In fact I see the behaviour that the screen first dims down and after quite a while gets brighter again. But it's nowhere near as dark that it would be unusable (in a room with no lights on, not in direct sunlight).

**Comment by kwah-gh** [comment]
I do have difficulty with dimmed screen in a daylight while in the room, and if I am outside (no direct sunlight either) screen appears almost black to me.

**Comment by UniversalSuperBox** [comment]
I can also confirm this issue. When waking up the device, it goes to minimum brightness and then turns the brightness up. This can also be seen when turning on automatic brightness for the first time. I'll speculate the sensor returns "very dark" when it's first turned on then gives a real data point later.

**Comment by vokimon** [comment]
Indeed this buggy behavior also happens with the preinstalled Android. So it could be the sensor. Yet, having this fixed in Ubuntu is a nice incentive to jump in.

**Comment by Flohack74** [comment]
I dont see we can fix this is the sensor comes up with unreliable data. We have to follow its values otherwise the whole function gets sluggish and makes no sense anymore. I would close this as a problem of the firmware of FP2.

**Comment by kwah-gh** [comment]
Only sensor? There maybe more to this, since sensor controlled brightness can get lower than  brightness level, which one gets on a minimal manual setting. I would expect these two should match each other at least at two points: 0% and 100%.

**Comment by FPfabi** [comment]
Hello,

having the same issue with FP2.

Device: FP2, New Camera Module
Channel: Edge
Build: 2020-W18, 16.04 LTS

in case other devices could have the same problem in the future, would it be solved by an additional feature "Min. Screen brigthness on wakeup" setting? E.g. UT OS takes the sensor reading and compares to the user specified minimum, after X sec. transitions to raw sensor value?

**Comment by Flohack74** [comment]
Well it seems the sensor reports that values, its not 100% sure if we can fix that. We rely on what we get from the Android sensors.
